crates/settlement-clients/starknet/src/tests/mod.rs

```diff
@@ -0,0 +1,239 @@
+        self
+    }
+
+    pub async fn wait_for_sync_to(&mut self, block_n: u64) -> &mut Self {
+        let rpc = self.json_rpc();
+        wait_for_cond(
+            || async {
+                match rpc.block_hash_and_number().await {
+                    Ok(got) => {
+                        if got.block_number < block_n {
+                            bail!("got block_n {}, expected {block_n}", got.block_number);
+                        }
+                        anyhow::Ok(())
+                    }
+                    Err(err) => bail!(err),
+                }
+            },
+            Duration::from_millis(5000),
+        )
+        .await;
+        self
+    }
+}
+
+impl Drop for MadaraCmd {
+    fn drop(&mut self) {
+        let Some(mut child) = self.process.take() else { return };
+        let kill = || {
+            let mut kill = Command::new("kill").args(["-s", "TERM", &child.id().to_string()]).spawn()?;
+            kill.wait()?;
+            anyhow::Ok(())
+        };
+        if let Err(_err) = kill() {
+            child.kill().unwrap()
+        }
+        child.wait().unwrap();
+    }
+}
+
+// this really should use unix sockets, sad
+
+const PORT_RANGE: Range<u16> = 19944..20000;
+
+struct AvailablePorts<I: Iterator<Item = u16>> {
+    to_reuse: Vec<u16>,
+    next: I,
+}
+
+lazy_static::lazy_static! {
+    static ref AVAILABLE_PORTS: Mutex<AvailablePorts<Range<u16>>> = Mutex::new(AvailablePorts { to_reuse: vec![], next: PORT_RANGE });
+}
+
+pub struct MadaraPortNum(pub u16);
+impl Drop for MadaraPortNum {
+    fn drop(&mut self) {
+        let mut guard = AVAILABLE_PORTS.lock().expect("poisoned lock");
+        guard.to_reuse.push(self.0);
+    }
+}
+
+pub fn get_port() -> MadaraPortNum {
+    let mut guard = AVAILABLE_PORTS.lock().expect("poisoned lock");
+    if let Some(el) = guard.to_reuse.pop() {
+        return MadaraPortNum(el);
+    }
+    MadaraPortNum(guard.next.next().expect("no more port to use"))
+}
+
+pub struct MadaraCmdBuilder {
+    args: Vec<String>,
+    env: HashMap<String, String>,
+    tempdir: TempDir,
+    port: MadaraPortNum,
+}
+
+impl Default for MadaraCmdBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl MadaraCmdBuilder {
+    pub fn new() -> Self {
+        Self {
+            args: Default::default(),
+            env: Default::default(),
+            tempdir: TempDir::with_prefix("madara-test").unwrap(),
+            port: get_port(),
+        }
+    }
+
+    pub fn args(mut self, args: impl IntoIterator<Item = impl Into<String>>) -> Self {
+        self.args = args.into_iter().map(Into::into).collect();
+        self
+    }
+
+    pub fn env(mut self, env: impl IntoIterator<Item = (impl Into<String>, impl Into<String>)>) -> Self {
+        self.env = env.into_iter().map(|(k, v)| (k.into(), v.into())).collect();
+        self
+    }
+
+    pub fn run(self) -> MadaraCmd {
+        let target_bin = option_env!("COVERAGE_BIN").unwrap_or("./target/debug/madara");
```

Comment:
let's just do an `expect` instead of a default value? will there be any case where the madara binary will be in `./target/debug/madara`